# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by scripts/Create-CopyPatch.
# 
# T2 SDE: package/*/embutils/xx-swapon.patch
# Copyright (C) 2006 - 2021 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

A tiny swapon I need for initrd.

  - Rene Rebe <rene@exactcode.de>

diff -urN embutils-0.17/readlink.c embutils-0.17-patched/readlink.c
--- embutils-0.17/swapon.c	1970-01-01 01:00:00.000000000 +0100
+++ embutils-0.17-patched/swapon.c	2005-12-19 00:13:23.000000000 +0100
@@ -0,0 +1,16 @@
+#include <stdio.h>
+#include <unistd.h>
+#include <sys/swap.h>
+
+int main(int argc,char *argv[]) {
+  int i;
+
+  for (i=1; i<argc; ++i) {
+    int n;
+    if (n = swapon(argv[i], 0)) {
+       fprintf(stderr, "Failed to activate swap\n");
+       return 1;
+    }
+  }
+  return 0;
+}
--- ./Makefile~	2006-11-05 11:54:48.000000000 +0100
+++ ./Makefile	2006-11-05 11:54:53.000000000 +0100
@@ -15,5 +15,5 @@
 printenv chrootuid renice \
-free losetup switch_root readlink
+free losetup switch_root readlink swapon
 
 OS:=$(shell uname)
 ifeq ($(OS),Linux)
